packages/app-mobile/components/screens/ShareManager/IncomingShareItem.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { ShareInvitation } from '@joplin/lib/services/share/reducer';
+import invitationRespond from '@joplin/lib/services/share/invitationRespond';
+import { Button, Card, Icon, Text } from 'react-native-paper';
+import { _ } from '@joplin/lib/locale';
+import { useCallback } from 'react';
+import { ViewStyle } from 'react-native';
+
+interface Props {
+	invitation: ShareInvitation;
+	processing: boolean;
+	containerStyle: ViewStyle;
+}
+
+const ShareIcon = (props: { size: number }) => <Icon {...props} source='account-arrow-left'/>;
+
+const IncomingShareItem: React.FC<Props> = props => {
+	const invitation = props.invitation;
+
+	const onAcceptInvitation = useCallback(() => {
+		void invitationRespond(invitation.id, invitation.share.folder_id, invitation.master_key, true);
+	}, [invitation]);
+	const onRejectInvitation = useCallback(() => {
+		void invitationRespond(invitation.id, invitation.share.folder_id, invitation.master_key, false);
+	}, [invitation]);
+
+	const sharer = invitation.share.user;
+	if (!sharer) return <Text>Error: Share missing user</Text>; // Should not happen
```

Comment:
```suggestion
```
This check might not be necessary. It was added because `invitation.share.user` is marked as optional in the type `StateShare`.